pkg/netpol/connlist/exposed_pods.go

```diff
@@ -0,0 +1,27 @@
+package connlist
+
+import "github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
+
+// ExposedPeer captures potential ingress and egress connections data for an exposed Peer
+type ExposedPeer interface {
+	// ExposedPeer is a peer for which the analysis found some potential exposure info
+	ExposedPeer() Peer
+	// IngressExposure list of the potential Ingress connections to the ExposedPeer
+	IngressExposure() []XgressExposureData
+	// EgressExposure list of the potential Egress connections from the ExposedPeer
+	EgressExposure() []XgressExposureData
+}
+
+// XgressExposureData data of potential connectivity for an existing peer to/from a representative peer
+type XgressExposureData interface {
+	// IsProtectedByNetpols Is the exposed peer protected by any xgress netpol
+	IsProtectedByNetpols() bool
+	// IsExposedToEntireCluster is the peer exposed to all namespaces in the cluster for the relevant direction
+	IsExposedToEntireCluster() bool
+	// NamespaceLabels the potential namespaces which the peer is exposed to by their labels
+	NamespaceLabels() map[string]string
+	// PodLabels the potential pods which the peer might be exposed to by labels
+	PodLabels() map[string]string
+	// PotentialConnectivity the potential connectivity of the exposure
+	PotentialConnectivity() common.AllowedConnectivity
```

Comment:
What combination of values is possible for these fields (each object is one instance in `IngressExposure()` or `EgressExposure()`) ?
How  is such an object going to be mapped to the visual connectivity output ? (what kind of edge is added and to which "new" node?)

Reply:
I don't think this is final, this may be changed during the computations, wanted to be aware of all data needed to be stored. 
( I am moving the `isProtected` flag out to the ExposedPod interface (splitted into 2 flags ingress+egress) - 

for case of non-protected peer (e.g. ingress not protected): ingress- exposure list will include one item with `exposedToEntireClass` and all-conns in the allowed connectivity >>
on the graph :  if the peer "free" (is not selected as "dst" for other peers in the graph (see question below)), will have an indication that it is exposed to any namespace (having a node for "any-namespace" and connected to it) 

but the classic case idea : 
let's assume we are having a pod `pod-a` which is selected by `ingress` netpols and having some potential conns (rules that has no match):
let's assume a rule with only-ns-selector, another with only-pod-selector and the third with both-ns-pod-selectors
it []ingress-exposure will be of len 3 :
ingress-exposure[0] will have non empty namespaceLabels with its allowed connectivity (protected, not exposed to entire cluster , empty podlabels)
ingress-exposure[1] will have non-empty podLabels
and ingress-exposure[2] will have non-empty nslabels and nonempty pod-labels 
in this case each item in the ingress-exposure list will have a node in the graph and edge to the `pod-a`

There are some questions to consider while computing: (maybe relevant to the "dot output" PR? or should be considered now?)
-> if the peer is protected (selected in ingress for example) and have an entire-cluster exposure (nsSelector : {} ), I think if there are other peers in the cluster with same connection , so automatically we will have ingress to it from other peers (should we also add a connection from "any namespace" node? )
-> same question if the peer is unprotected but have conns from other peers in the cluster?
-> if we have rule that exposes the peer to entire cluster and other rules with new-potential peers , i think we don't need to add node for the potential peers only the "any-namespace" node


- if we shouldn't add conns to/from entire cluster if there are already existing conns with other peers - I might also add a flags to indicate if there is connections between a peer and other peers in the cluster (with same allowed conns)

Reply:
- if a peer is exposed to entire cluster , it will have a representation of the connection/s with the entire cluster (i.e. "any-namespace" node) (even if it is connected to other peers in the cluster)
-  if we have rule that exposes the peer to entire cluster and other rules with new-potential peers - only the entire-cluster rule is considered (the most general connection)
- the new nodes types :
   - any-namespace
   - any namespace with labels
   - any pod with labels in the netpol's namespace
   - any pod with labels in any-namespace
   - any pod with labels in any namespace with labels